### ISSUE TITLE
Improve snowflake_stages (Data Source) to detect external/internal stages

### ISSUE THREAD

**Issue by TerjeRusska** [Snowflake-Labs/terraform-provider-snowflake#1544]
**Is your feature request related to a problem? Please describe.**
Previous issue that relates to the request: https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/772

It is not possible combine [snowflake_stage_grant (Resource)](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/resources/stage_grant) with [snowflake_stages (Data Source)](https://registry.terraform.io/providers/Snowflake-Labs/snowflake/latest/docs/data-sources/stages) by looping through the resources since USAGE only applies to external stages, READ | WRITE only applies to internal stages.

https://docs.snowflake.com/en/sql-reference/sql/grant-privilege#usage-notes

**Describe the solution you'd like**

Add the following attributes to the nested schema for stages in snowflake_stages (Data Source):

- type

`show stages in account;`

**Describe alternatives you've considered**

Currently have to manually split the stages

**Additional context**

**Comment by sfc-gh-swinkler** [comment]
Yes this makes sense, and I agree we should do this. Could you please file a support ticket with your Snowflake account representative on this issue? It helps a lot in getting prioritization for this.

**Comment by sfc-gh-asawicki** [comment]
Hey, this issue is part of the work planned for the next weeks. See more context in the [new roadmap entry](https://github.com/snowflakedb/terraform-provider-snowflake/blob/main/ROADMAP.md#2025-12-18-roadmap-update).
